charts/memgraph-high-availability/templates/coordinators.yaml

```diff
@@ -9,28 +9,85 @@ spec:
   selector:
     matchLabels:
       app: memgraph-coordinator-{{ $coordinator.id }}
+      role: coordinator
   template:
     metadata:
       labels:
         app: memgraph-coordinator-{{ $coordinator.id }}
+        role: coordinator
         instance-type: coordinator
+
     spec:
-    {{ if $.Values.memgraph.affinity.enabled }}
+      {{- if $.Values.memgraph.affinity.enabled }}
       affinity:
+        {{- if $.Values.memgraph.affinity.nodeSelection }}
+        # Node Selection Affinity: Scheduled on nodes with specific label key and value
+        nodeAffinity:
+          requiredDuringSchedulingIgnoredDuringExecution:
+            nodeSelectorTerms:
+            - matchExpressions:
+              - key: {{ $.Values.memgraph.affinity.roleLabelKey }}
+                operator: In
+                values:
+                - {{ $.Values.memgraph.affinity.coordinatorNodeLabelValue }}
+        podAntiAffinity :
+          requiredDuringSchedulingIgnoredDuringExecution:
+          - labelSelector:
+              matchExpressions:
+              - key: role
+                operator: In
+                values:
+                - coordinator
```

Comment:
Do we need podAntiAffinity and if yes, why?

Reply:
Resolved on the call: podAntiAffinity is needed to avoid scheduling multiple coordinators on the same node